mailet/api/src/main/java/org/apache/mailet/AttributeValue.java

```diff
@@ -0,0 +1,205 @@
+    }
+
+    public static AttributeValue<String> of(String value) {
+        return new AttributeValue<>(value, Serializer.STRING_SERIALIZER);
+    }
+
+    public static AttributeValue<Integer> of(Integer value) {
+        return new AttributeValue<>(value, Serializer.INT_SERIALIZER);
+    }
+
+    public static AttributeValue<Long> of(Long value) {
+        return new AttributeValue<>(value, Serializer.LONG_SERIALIZER);
+    }
+
+    public static AttributeValue<Float> of(Float value) {
+        return new AttributeValue<>(value, Serializer.FLOAT_SERIALIZER);
+    }
+
+    public static AttributeValue<Double> of(Double value) {
+        return new AttributeValue<>(value, Serializer.DOUBLE_SERIALIZER);
+    }
+
+    public static AttributeValue<QueueSerializable> of(QueueSerializable value) {
+        return new AttributeValue<>(value, Serializer.QUEUE_SERIALIZABLE_SERIALIZER);
+    }
+
+    public static AttributeValue<URL> of(URL value) {
+        return new AttributeValue<>(value, Serializer.URL_SERIALIZER);
+    }
+
+    @SuppressWarnings({ "unchecked", "rawtypes" })
+    public static AttributeValue<Collection<AttributeValue<?>>> of(Collection<AttributeValue<?>> value) {
+        return new AttributeValue<>(value, new Serializer.CollectionSerializer());
+    }
+
+    @SuppressWarnings({ "unchecked", "rawtypes" })
+    public static AttributeValue<Map<String, AttributeValue<?>>> of(Map<String, AttributeValue<?>> value) {
+        return new AttributeValue<>(value, new Serializer.MapSerializer());
+    }
+
+    public static AttributeValue<Serializable> ofSerializable(Serializable value) {
+        return new AttributeValue<>(value, new Serializer.FSTSerializer());
+    }
+
+    @SuppressWarnings("unchecked")
+    public static AttributeValue<?> ofAny(Object value) {
+        if (value instanceof Boolean) {
+            return of((Boolean) value);
+        }
+        if (value instanceof String) {
+            return of((String) value);
+        }
+        if (value instanceof Integer) {
+            return of((Integer) value);
+        }
+        if (value instanceof Long) {
+            return of((Long) value);
+        }
+        if (value instanceof Float) {
+            return of((Float) value);
+        }
+        if (value instanceof Double) {
+            return of((Double) value);
+        }
+        if (value instanceof Collection<?>) {
+            return of(((Collection<AttributeValue<?>>) value));
+        }
+        if (value instanceof Map<?,?>) {
+            return of(((Map<String, AttributeValue<?>>) value));
+        }
+        if (value instanceof QueueSerializable) {
+            return of((QueueSerializable) value);
+        }
+        if (value instanceof URL) {
+            return of((URL) value);
+        }
+        if (value instanceof Serializable) {
+            return ofSerializable((Serializable) value);
+        }
+        throw new IllegalArgumentException("input should at least be Serializable");
```

Comment:
Could you add some information about what kind of objects we support? I think the message could be more helpful